samples/matter/common/src/board.cpp

```diff
@@ -0,0 +1,266 @@
+void Board::FunctionTimerTimeoutCallback(k_timer *timer)
+{
+	TaskExecutor::PostTask([] { FunctionTimerEventHandler(); });
+}
+
+void Board::FunctionTimerEventHandler()
+{
+	/* If we reached here, the button was held past FactoryResetTriggerTimeout, initiate factory reset */
+	if (sInstance.mFunction == BoardFunctions::SoftwareUpdate) {
+		LOG_INF("Factory reset has been triggered. Release button within %ums to cancel.",
+			FactoryResetConsts::kFactoryResetTriggerTimeout);
+
+		/* Start timer for kFactoryResetCancelWindowTimeout to allow user to cancel, if required. */
+		sInstance.StartTimer(FactoryResetConsts::kFactoryResetCancelWindowTimeout);
+		sInstance.mFunction = BoardFunctions::FactoryReset;
+
+		/* Turn off all LEDs before starting blink to make sure blink is coordinated. */
+		sInstance.ResetAllLeds();
+
+		sInstance.mLED1.Blink(LedConsts::kBlinkRate_ms);
+		sInstance.mLED2.Blink(LedConsts::kBlinkRate_ms);
+#if NUMBER_OF_LEDS == 3
+		sInstance.mLED3.Blink(LedConsts::kBlinkRate_ms);
+#elif NUMBER_OF_LEDS == 4
+		sInstance.mLED3.Blink(LedConsts::kBlinkRate_ms);
+		sInstance.mLED4.Blink(LedConsts::kBlinkRate_ms);
+#endif
+	} else if (sInstance.mFunction == BoardFunctions::FactoryReset) {
+		/* Actually trigger Factory Reset */
+		sInstance.mFunction = BoardFunctions::None;
+		chip::Server::GetInstance().ScheduleFactoryReset();
+
+	}
+}
+
+void Board::ButtonEventHandler(ButtonState buttonState, ButtonMask hasChanged)
+{
+	if (BLUETOOTH_ADV_BUTTON_MASK & hasChanged) {
+		ButtonAction action =
+			(BLUETOOTH_ADV_BUTTON_MASK & buttonState) ? ButtonAction::Pressed : ButtonAction::Released;
+		TaskExecutor::PostTask([action] { StartBLEAdvertisementHandler(action); });
+	}
+
+	if (FUNCTION_BUTTON_MASK & hasChanged) {
+		ButtonAction action =
+			(BLUETOOTH_ADV_BUTTON_MASK & buttonState) ? ButtonAction::Pressed : ButtonAction::Released;
+		TaskExecutor::PostTask([action] { FunctionHandler(action); });
+	}
+}
+
+void Board::FunctionHandler(const ButtonAction &action)
+{
+	if (action == ButtonAction::Pressed) {
+		if (!sInstance.mFunctionTimerActive && sInstance.mFunction == BoardFunctions::None) {
+			sInstance.mFunction = BoardFunctions::SoftwareUpdate;
+			sInstance.StartTimer(FactoryResetConsts::kFactoryResetTriggerTimeout);
+		}
+	} else {
+		/* If the button was released before factory reset got initiated, trigger a software update. */
+		if (sInstance.mFunctionTimerActive && sInstance.mFunction == BoardFunctions::SoftwareUpdate) {
+			sInstance.CancelTimer();
+			sInstance.mFunction = BoardFunctions::None;
+		} else if (sInstance.mFunctionTimerActive && sInstance.mFunction == BoardFunctions::FactoryReset) {
+			sInstance.ResetAllLeds();
+			sInstance.CancelTimer();
+			sInstance.mLedStateHandler();
+			sInstance.mFunction = BoardFunctions::None;
+			LOG_INF("Factory reset has been canceled");
+		}
+	}
+}
+
+void Board::StartBLEAdvertisementHandler(const ButtonAction &action)
+{
+#ifndef CONFIG_NCS_SAMPLE_MATTER_CUSTOM_BLUETOOTH_ADVERTISING
```

Comment:
Do I miss something or we still need to set `NCS_SAMPLE_MATTER_CUSTOM_BLUETOOTH_ADVERTISING` in the light_switch sample and implement the custom advertising accordingly? Without that, we mishandle the UI in case of the nRF7002 DK. I guess for that we also need to make the `StartBLEAdvertisement` a public method.

Reply:
Add let's document the StartBLEAdvertising as a part of API to mention that it can be used for implementing a custom BLE advertising trigger :)